File: monotonic-array/MonotonicArray.swift
import Foundation

class MonotonicArray {
    func isMonotonic(array: [Int]) -> Bool {
        if(array.isEmpty) { return true }
            var isIncreasing = array[0] < array[array.count - 1] 
            for index in 0 ..< array.count - 1 {
            if(isIncreasing && (array[index] > array[index + 1])) { return false }
            if(!isIncreasing && (array[index] < array[index + 1])) { return false }
        }
        return true
    }
}

File: monotonic-array/MonotonicArray.java
public class MonotonicArray {
    public boolean isMonotonic(int[] array) {
        if(array.length == 0) return true;
        boolean isIncreasing = array[0] < array[array.length - 1];
        for(int index = 0; index < array.length - 1; index++) {
            if(isIncreasing && (array[index]) > array[index + 1]) return false;
            if(!isIncreasing && (array[index]) < array[index + 1]) return false;
        }
        return true;
    }
}
